### ISSUE TITLE
Issue running `upload_dbt_artifacts` in production with dbt Cloud orchestrator

### ISSUE THREAD

**Issue by lewisarmistead** [brooklyn-data/dbt_artifacts#13]
Hi there - this package fits our needs for monitoring model run times perfectly, and we were able to run all aspects of it locally using the dbt CLI and in our development environments on dbt Cloud using the dbt Cloud IDE; however, we're unable to get the `dbt-run-operation upload_dbt_artifacts --args '{filenames: [manifest, run_results]}'` to run as a step in a dbt Cloud job due to this recurring error:

```
Running with dbt=0.19.1
Clearing existing files from Stage:
    remove @ANALYTICS.dbt_artifacts.artifacts pattern='.*.json.gz';

Encountered an error while running operation: Database Error
  002003 (02000): SQL compilation error:
  Stage 'ANALYTICS.DBT_ARTIFACTS.ARTIFACTS' does not exist or not authorized.
```

Some troubleshooting steps we've taken so far:
- I ensured that the user & role associated with the environments that we've tried running this from in dbt Cloud have access to the table (I've even tried granting all privileges and ownership to the roles being used) and usage privileges on the schema where it's located
- I tried running using various environments including our production environment, our pull request environment configured for GitHub CI checks, and some temporary environments I made to try to emulate my local development environment and my dbt Cloud development environment
- I tried manually specifying the location of the artifacts in the `dbt_project.yml` file and reinstalling the package
- I tried using dbt versions 0.19.1 and 0.19.0 (also 0.18.0, but that failed as expected due to the artifacts version)
- I tried running this using the `dev` and `prod` targets that we use internally as well as the `default` target

All of those steps result in the same failure message shown above only when running the operation from a dbt Cloud job.

I've also tried using our dbt Cloud user's credentials, which we typically only use for dbt Cloud jobs, and was able to get the artifact upload to work properly when running locally.

Do you have any ideas on why we might be seeing this failure or other steps I could try to troubleshoot? I've tried doing most of the troubleshooting on uploads following some sort of `dbt run` operation (those run operations have succeeded). I am also able to run the artifact models from a dbt Cloud job successfully, it only seems to be the uploads that are causing issues.

Any advice would be appreciated - thanks for the help!

**Comment by kon-mal** [comment]
Hello,

Is it possible that you're running your dbt cloud jobs on a different database/schema than locally? This would mean that you need to initially run `dbt run-operation create_artifact_resources` to create the stage in the table to which you tried to upload your data ('ANALYTICS.DBT_ARTIFACTS.ARTIFACTS').

You can check here that this command is first creating the schema and then the necessary stage (https://github.com/tailsdotcom/dbt_artifacts/blob/main/macros/create_artifact_resources.sql)

Just a thought! We've been running this command (`dbt run-operation upload_dbt_artifacts --args '{filenames: [manifest, run_results]}`) in dbt cloud for the last few weeks and it worked like a charm.

**Comment by lewisarmistead** [comment]
Thanks for the response, @kon-mal - this helped me fix the issue!

The dbt Cloud jobs we're running are using the same Snowflake role that is successfully running dbt commands in our other production dbt Cloud jobs. 

I tried recreating the ANALYTICS.DBT_ARTIFACTS.ARTIFACTS table under the same role using that `dbt run-operation create_artifact_resources` command, but I was only getting permissions errors on the Snowflake **STAGE** ANALYTICS.DBT_ARTIFACTS.ARTIFACTS from a dbt Cloud job - I didn't realize there was a difference between the *TABLE* ANALYTICS.DBT_ARTIFACTS.ARTIFACTS, which the role the dbt Cloud job was using had full permissions on, vs. the *STAGE* of the same name, which I had originally created using another role. 

My lack of understanding the difference between the TABLE and STAGE Snowflake objects was the root of the issue here 😞. That explains why we *were* able to run the `dbt run -m dbt_artifacts` commands in our dbt Cloud jobs, which were reading from the artifacts TABLE, and *were not* able to run not any of the write artifacts commands in this package, which were using that STAGE object.

I wound up dropping both the artifacts STAGE and TABLE and recreating them using a dbt Cloud job to ensure that the role had permissions to both - I had tried something similar previously but only dropped the TABLE and not the STAGE, which is why that wasn't working.

It even mentions that both a stage and table are created during the `dbt run-operation create_artifact_resources` command, but I didn't pick up on that. As a friendly suggestion to the package maintainers, it may be worth highlighting this difference between STAGE and TABLE for other Snowflake novices like myself in case there are others that test out this package using a different role than the one used in their production jobs.

Thanks again for the help! 👍
